File: springboot-gradle/src/main/java/com/example/springbootgradle/repository/BookRepository.java
package com.example.springbootgradle.repository;

/**
 * Created by leil7 on 2016/11/22.
 */

import java.util.List;

import com.example.springbootgradle.bean.Book;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BookRepository extends JpaRepository<Book, Long> {

    List<Book> findByTitle(String reader);

}

File: springboot/springboot-kotlin/src/main/java/com/example/springbootkotlin/repository/BookRepository.kt
package com.example.springbootkotlin.repository

/**
 * Created by leil7 on 2016/11/22.
 */

import com.example.springbootkotlin.bean.Book
import org.springframework.data.jpa.repository.JpaRepository

interface BookRepository : JpaRepository<Book, Long> {

    fun findByTitle(reader: String): List<Book>

}